File: src/main/java/com/jelly/MightyMiner/baritone/automine/structures/Node.java
package com.jelly.MightyMiner.baritone.automine.structures;

import net.minecraft.util.BlockPos;

public class Node {
    public Node(BlockPos pos){
        this.pos = pos;
    }

    public BlockPos pos;

    public double hCost;
    public double gCost = -1;
    public double fCost;
    public Node parentNode;
}

File: src/main/java/com/jelly/MightyMinerV2/Pathfinder/Node.kt
package com.jelly.MightyMinerV2.Pathfinder

import net.minecraft.util.BlockPos

data class Node(
    val pos: BlockPos,
    val parent: Node?,
    val gCost: Double, // Cost from start to this node
    val hCost: Double  // Heuristic cost to the end node
) {
    val fCost: Double get() = gCost + hCost
}

